java/java.lsp.server/vscode/src/extension.ts

```diff
@@ -705,13 +732,38 @@ class NetBeansConfigurationProvider implements vscode.DebugConfigurationProvider
                     request: "attach",
                 };
                 for (let i = 0; i < ac.arguments.length; i++) {
-                    debugConfig[ac.arguments[i]] = ac.defaultValues[i];
+                    let defaultValue: string = ac.defaultValues[i];
+                    if (!defaultValue.startsWith("${command:")) {
+                        // Create a command that asks for the argument value:
+                        let cmd: string = "java.attachDebugger.connector." + ac.id + "." + ac.arguments[i];
+                        debugConfig[ac.arguments[i]] = "${command:" + cmd + "}";
+                        if (!commandValues.has(cmd)) {
+                            commandValues.set(cmd, ac.defaultValues[i]);
+                            let description: string = ac.descriptions[i];
+                            context.subscriptions.push(commands.registerCommand(cmd, async (ctx) => {
+                                return vscode.window.showInputBox({
+                                    prompt: description,
+                                    value: commandValues.get(cmd),
+                                }).then((value) => {
+                                    if (value) {
+                                        commandValues.set(cmd, value);
```

Comment:
I can see that the `commandValues` storage receives user input. Is the InputBox displayed each time the `debugConfig` is launched (although pre-filled with the previous input) ?

Reply:
Yes. If we would not ask again, the user would have no way to change the values.
If you attach often to some particular destination, you should put it into the launch.json, IMHO.
I tried to do the best with what we can do in VSCode, but suggestions are welcomed, of course.